protobuf-maven-plugin/src/main/java/io/github/ascopes/protobufmavenplugin/utils/HostSystem.java

```diff
@@ -54,7 +54,9 @@ public final class HostSystem {
 
   @Inject
   public HostSystem() {
-    this(System.getProperties(), System.getenv());
+    // GH-271: Pass a map with the required environment variables values
+    this(System.getProperties(), Map.of("PATH", System.getenv("PATH"),
```

Comment:
Would passing `Function<String, String>` be easier here?

That way you could pass System::getenv directly and preserve the behaviour that is needed and provided by the `System.getenv(String)` overload but not the 0-arg overload, without needing to hardcode the variable names in two places

Reply:
Yes, that would work, but I was going for minimally invasive change. `System.getenv(String)` can return `null`, and the direct usages of the overloaded constructor would need to be updated.
However, if this is the direction you want to go, it is perfectly fine with me.

Reply:
Think it would be good, it should also help with some of the failures due to missing variables on non-Windows

Reply:
Ok, updated.